app/controllers/posts_controller.rb

```diff
@@ -1,69 +1,19 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: %i[ show edit update destroy ]
 
-  # GET /posts or /posts.json
-  def index
-    @posts = Post.all
-  end
-
-  # GET /posts/1 or /posts/1.json
-  def show
-  end
-
-  # GET /posts/new
   def new
     @post = Post.new
   end
 
-  # GET /posts/1/edit
-  def edit
-  end
-
-  # POST /posts or /posts.json
   def create
-    @post = Post.new(post_params)
-
-    respond_to do |format|
-      if @post.save
-        format.html { redirect_to @post, notice: "Post was successfully created." }
-        format.json { render :show, status: :created, location: @post }
-      else
-        format.html { render :new, status: :unprocessable_entity }
-        format.json { render json: @post.errors, status: :unprocessable_entity }
-      end
+    if @post.invalid?
+      render "new"
+    else
+      render "show"
```

Comment:
In this case, we want to redirect. The `show` action is responsible for handling this.